libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteControllerFacade.kt

```diff
@@ -0,0 +1,138 @@
+import com.mapbox.navigation.base.internal.utils.mapToSdkRouteOrigin
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.toDirectionsRoutes
+import com.mapbox.navigation.base.route.toNavigationRoutes
+import com.mapbox.navigator.RerouteError
+import com.mapbox.navigator.RerouteObserver
+import com.mapbox.navigator.RouteInterface
+import com.mapbox.navigator.RouterOrigin
+import java.net.URL
+import java.util.concurrent.CopyOnWriteArraySet
+
+internal class MapboxRerouteControllerFacade private constructor(
+    private val rerouteObserver: RerouteControllersManager.Observer,
+    private val nativeRerouteController: NativeExtendedRerouteControllerInterface,
+) : NavigationRerouteController {
+
+    override var state: RerouteState = RerouteState.Idle
+        @VisibleForTesting
+        internal set(value) {
+            if (value == field) return
+            field = value
+            observers.forEach { it.onRerouteStateChanged(field) }
+        }
+
+    private val observers = CopyOnWriteArraySet<RerouteController.RerouteStateObserver>()
+
+    internal companion object {
+        private const val LOG_CATEGORY = "MapboxRerouteControllerFacade"
+
+        internal operator fun invoke(
+            rerouteObserver: RerouteControllersManager.Observer,
+            nativeRerouteController: NativeExtendedRerouteControllerInterface,
+        ): MapboxRerouteControllerFacade = MapboxRerouteControllerFacade(
+            rerouteObserver,
+            nativeRerouteController,
+        )
+    }
+
+    init {
+        nativeRerouteController.addRerouteObserver(object : RerouteObserver {
+            override fun onRerouteDetected(routeRequest: String): Boolean {
+                state = RerouteState.FetchingRoute
+                return true
+            }
+
+            override fun onRerouteReceived(
+                routeResponse: String,
+                routeRequest: String,
+                origin: RouterOrigin
+            ) {
+                state = RerouteState.RouteFetched(origin.mapToSdkRouteOrigin())
+                rerouteObserver.onNewRoutes(
+                    DirectionsResponse.fromJson(
+                        routeResponse, RouteOptions.fromUrl(URL(routeRequest))
+                    )
+                        .routes().toNavigationRoutes(origin.mapToSdkRouteOrigin()),
+                    RerouteControllersManager.Observer.Reason.Reroute,
+                )
+                state = RerouteState.Idle
+            }
+
+            override fun onRerouteCancelled() {
+                state = RerouteState.Interrupted
+                state = RerouteState.Idle
+            }
+
+            override fun onRerouteFailed(error: RerouteError) {
+                state = RerouteState.Failed(error.message)
+                state = RerouteState.Idle
+            }
+
+            override fun onSwitchToAlternative(route: RouteInterface) {
+                rerouteObserver.onNewRoutes(
```

Comment:
@DenisPryt could you help with it? 
- I cannot catch `swithToAlternative` even when changing the custom config of alternatives'`traveledDistanceBeforeRemove` to `150.0`
- I don't see in logs if custom config was applied, earlier I saw it in logs. Maybe it was changed in some version of NN?